sdk/src/Core/Amazon.Runtime/Pipeline/ErrorHandler/ErrorHandler.cs

```diff
@@ -202,7 +202,11 @@ private bool ProcessException(IExecutionContext executionContext, Exception exce
                 {
                     return exceptionHandler.Handle(executionContext, exception);
                 }
-                exceptionType = exceptionType.BaseType;
+
+                if (exceptionType != typeof(Exception))
```

Comment:
I think this logic would be easier to follow if you replace this `if` statement and added ` && exceptionType != typeof(object)` into the while condition. Otherwise it looks weird to me to have the `exceptionType != typeof(Exception)` twice.

Reply:
Done.